modules/appeals_api/app/controllers/appeals_api/v2/decision_reviews/legacy_appeals_controller.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+require 'caseflow/service'
+require 'common/exceptions'
+
+class AppealsApi::V2::DecisionReviews::LegacyAppealsController < AppealsApi::ApplicationController
+  SSN_REGEX = /^[0-9]{9}$/.freeze
+
+  skip_before_action(:authenticate)
+  before_action :validate_headers, only: %i[index]
+
+  EXPECTED_HEADERS = %w[X-VA-SSN X-VA-File-Number].freeze
+
+  UNUSABLE_RESPONSE_ERROR = {
+    errors: [
+      {
+        title: 'Bad Gateway',
+        code: 'bad_gateway',
+        detail: 'Received an unusable response from Caseflow.',
+        status: 502
+      }
+    ]
+  }.freeze
+
+  def index
+    get_legacy_appeals_from_caseflow
+
+    if caseflow_response_usable?
+      render_response(caseflow_response)
+    else
+      render_unusable_response_error
+    end
+  end
+
+  private
+
+  attr_reader :caseflow_response, :backend_service_exception
+
+  def request_headers
+    EXPECTED_HEADERS.reduce({}) do |hash, key|
+      hash.merge(key => request.headers[key])
+    end.compact
+  end
+
+  def validate_headers
+    validation_errors = []
+    ssn = request.headers['X-VA-SSN']
+    file_number = request.headers['X-VA-File-Number']
+
+    if ssn.nil? && file_number.nil?
+      validation_errors << { status: 422, detail: 'X-VA-SSN or X-VA-File-Number is required' }
+    end
+
+    if ssn.present? && !SSN_REGEX.match?(ssn)
+      validation_errors << { status: 422, detail: "X-VA-SSN has an invalid format. Pattern: #{SSN_REGEX.inspect}" }
+    end
+
+    render_validation_errors(validation_errors)
+  end
+
+  def render_validation_errors(validation_errors)
+    return if validation_errors.empty?
+
+    render json: { errors: validation_errors }, status: :unprocessable_entity unless validation_errors.empty?
```

Comment:
You've already guarded against empty validation errors above, no need to do it a second time with the `unless`.

Reply:
derp